community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/codegen/ir/Instruction.scala

```diff
@@ -34,7 +34,7 @@ trait Instruction {
   // Aggregating methods -- final to prevent overriding
   final def allOperatorIds: Set[String] = treeView.flatMap(_.operatorId).toSet
 
-  final def allColumns: Set[String] = treeView.flatMap(_.columnNames).toSet
+  final def allColumns: Seq[String] = treeView.flatMap(_.columnNames)
```

Comment:
Could this not have been enough?

And if this is not enough, is it required? 
It looks like this could have been a first attempt that didn't work and then the problem was solved in a different way. If that is the case, perhaps this method isn't needed anymore and should be removed?